server/task_check_executor_statement_advisor_composite.go

```diff
@@ -50,11 +50,22 @@ func (exec *TaskCheckStatementAdvisorCompositeExecutor) Run(ctx context.Context,
 		return nil, common.Errorf(common.Invalid, fmt.Errorf("invalid check statement advise payload: %w", err))
 	}
 
-	policy, err := server.store.GetSchemaReviewPolicyByID(ctx, payload.PolicyID)
+	policy, err := server.store.GetSchemaReviewPolicyNormalByID(ctx, payload.PolicyID)
 	if err != nil {
 		return nil, common.Errorf(common.Internal, fmt.Errorf("failed to get schema review policy: %w", err))
 	}
 
+	if policy.Empty() {
```

Comment:
Can we keep the `GetSchemaReviewPolicyByID` and just check if the policy is archived here? Seems `GetSchemaReviewPolicyNormalByID` returns a empty policy if not found is weird

Reply:
But `GetScemaReviewPolicyByID` will return the archived policy.

Reply:
`GetSchemaReviewPolicyByID` will return the `api.SchemaReviewPolicy` which we can't distinguish between `normal` and `archived`.

Reply:
Or maybe return nil if the normal policy not found?